internal/kafka/internal/services/quota/quota_management_list_service_test.go

```diff
@@ -462,7 +461,45 @@ func Test_QuotaManagementListReserveQuota(t *testing.T) {
 				InstanceType:   tt.args.instanceType.String(),
 			}
 			_, err := quotaService.ReserveQuota(kafka, tt.args.instanceType)
-			gomega.Expect(tt.wantErr).To(gomega.Equal(err))
+			Expect(tt.wantErr).To(Equal(err))
+		})
+	}
+}
+func Test_DefaultQuotaServiceFactory_GetQuotaService(t *testing.T) {
+	type fields struct {
+		QuoataServiceContainer map[api.QuotaType]services.QuotaService
+	}
+	type args struct {
+		quoataType api.QuotaType
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   services.QuotaService
+		want1  *errors.ServiceError
+	}{
+		{
+			name: "Should return nil and error if QuotaType is invalid",
+			fields: fields{
+				QuoataServiceContainer: map[api.QuotaType]services.QuotaService{},
+			},
+			args: args{
+				quoataType: api.UndefinedQuotaType,
+			},
+			want:  nil,
+			want1: errors.GeneralError("invalid quota service type: %v", api.QuotaManagementListQuotaType),
+		},
+	}
+	RegisterTestingT(t)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			factory := &DefaultQuotaServiceFactory{
+				quotaServiceContainer: map[api.QuotaType]services.QuotaService{},
+			}
+			got, got1 := factory.GetQuotaService(tt.args.quoataType)
+			Expect(got).To(BeNil())
+			Expect(got1).To(Equal(tt.want1))
```

Comment:
This is incorrect. Same issue as in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/981#discussion_r862971406.
Additionally please review all the similar checks added in this PR to make sure all of them are fixed.